flow_process_components/FlowBaseComponents/force-app/main/default/classes/fbc_ExtractMetadataStringFromZipfile.cls

```diff
@@ -0,0 +1,84 @@
+/**
+ * Eric Smith - August 2020
+ * 
+ *      Extract a single metadata file from the Zip file returned by the Retrieve
+ *      method of the fbc_TransferMetada LWC
+ * 
+ *      Input:  Zip File String
+ * 
+ *      Output: Extracted File String
+ *              Escaped File String (Angle brackets replaced by &lt and &gt)
+ * 
+ */
+
+public with sharing class fbc_ExtractMetadataStringFromZipfile {
+
+    @invocableMethod
+    public static List<Result> extract(List<Request> requests) {
+        List<Result> responseWrapper = new List<Result>();
+
+        for (Request curRequest : requests) {
+            Result curResult = new Result();
+            String unescapedString = unpackZipfile(curRequest.flowMetadataZipFile);
+            curResult.extractedFlowMetadata = unescapedString;
+            curResult.escapedFlowMetadata = escape(unescapedString);
+            responseWrapper.add(curResult);
+        }
+
+        return responseWrapper;
+
+    }
+
+    public static String unpackZipfile(String metadataZipfile) {
```

Comment:
is this the same unpack zipfile as the one in Retrieve Metadata? Can we DRY these up?

Reply:
would it possibly make sense to add the unpack capability to the RetrieveMetadata class? There's an argument that it's a fit...